pkg/kubelet/pod_workers.go

```diff
@@ -106,8 +122,10 @@ func (p *podWorkers) UpdatePod(pod *api.BoundPod, updateComplete func()) {
 	p.podLock.Lock()
 	defer p.podLock.Unlock()
 	if podUpdates, exists = p.podUpdates[uid]; !exists {
-		// Currently all update request for a given pod coming when another
-		// update of this pod is being processed are ignored.
+		// We need to have a buffer here, because checkForUpdates() method that
```

Comment:
I'm not a huge fan of this since it seems brittle to future changes. I don't have a better option right now though so I'm just complaining :) Don't block the PR on this.